src/ResourceManager/RecoveryServices.Backup/Commands.RecoveryServices.Backup.Providers/Providers/AzureSqlPsBackupProvider.cs

```diff
@@ -67,12 +72,44 @@ public Management.RecoveryServices.Backup.Models.ProtectedItemResponse GetProtec
 
         public AzureRmRecoveryServicesBackupRecoveryPointBase GetRecoveryPointDetails()
         {
-            throw new NotImplementedException();
+            AzureRmRecoveryServicesBackupAzureSqlItem item = ProviderData.ProviderParameters[GetRecoveryPointParams.Item]
+                as AzureRmRecoveryServicesBackupAzureSqlItem;
+
+            string recoveryPointId = ProviderData.ProviderParameters[GetRecoveryPointParams.RecoveryPointId].ToString();
+
+            Dictionary<UriEnums, string> uriDict = HelperUtils.ParseUri(item.Id);
+            string containerUri = HelperUtils.GetContainerUri(uriDict, item.Id);
+            string protectedItemName = HelperUtils.GetProtectedItemUri(uriDict, item.Id);
+
+            var rpResponse = HydraAdapter.GetRecoveryPointDetails(containerUri, protectedItemName, recoveryPointId);
+            return RecoveryPointConversions.GetPSAzureRecoveryPoints(rpResponse, item);
         }
 
         public List<AzureRmRecoveryServicesBackupRecoveryPointBase> ListRecoveryPoints()
         {
-            throw new NotImplementedException();
+            DateTime startDate = (DateTime)(ProviderData.ProviderParameters[GetRecoveryPointParams.StartDate]);
+            DateTime endDate = (DateTime)(ProviderData.ProviderParameters[GetRecoveryPointParams.EndDate]);
+            AzureRmRecoveryServicesBackupAzureSqlItem item = ProviderData.ProviderParameters[GetRecoveryPointParams.Item]
+                as AzureRmRecoveryServicesBackupAzureSqlItem;
+
+            Dictionary<UriEnums, string> uriDict = HelperUtils.ParseUri(item.Id);
+            string containerUri = HelperUtils.GetContainerUri(uriDict, item.Id);
+            string protectedItemName = HelperUtils.GetProtectedItemUri(uriDict, item.Id);
+
+            TimeSpan duration = endDate - startDate;
+            if (duration.TotalDays > 30)
+            {
+                throw new Exception(Resources.RestoreDiskTimeRangeError); //tbd: Correct nsg and exception type
+            }
```

Comment:
Check with PM if its need for you or not. In iaas vm we added because pagination is not supported by service